### ISSUE TITLE
Config:

### ISSUE THREAD

**Issue by afilini** [romanz/electrs#411]
**Have you read the documentation?**
Yes.

**How did you configure electrs?**

<details>
<summary>/etc/electrs/config.toml</summary>

```
cookie=...
```

</details>

<details>
<summary>local_config.toml</summary>

```
# yes, this is empty
```

</details>

Arguments: `--conf ./local_config.toml --cookie-file ...`

**Debug output**
```
WARNING: cookie option is deprecated and will be removed in the future!

You most likely want to use cookie_file instead.
If you really don't want to use cookie_file for a good reason and knowing the consequences use the auth option
See authentication section in electrs usage documentation for more details.
https://github.com/romanz/electrs/blob/master/doc/usage.md#configuration-files-and-priorities
Error: ambigous configuration - cookie and cookie_file can't be specified at the same time
```

**Expected behavior**

`electrs` should override the options from the "system" config file in `/etc` with the options from the local config file, which is empty.

From the usage doc:

> There are two special arguments --conf which reads the specified file and --conf-dir, which read all the files in the specified directory.
> The options in those files override everything that was set previously, including arguments that were passed before these two special arguments.

Looking at the code, it looks like those files do override the options, but only if the higher-priority conf file contains the same option. Basically, it will only add more options from the newer files, never remove them. So if the system conf file contains a `cookie` option there's no way for me to remove it unless I physically remove that file.

**Comment by Kixunil** [comment]
Heh, I knew the deprecation implementation was imperfect but didn't expect anyone to hit this edge case. :) Although I believe it's fair to say that you probably should clean up your config off deprecated option.

I was putting off implementing enums in `configure_me` but your use case is quite motivating. How serious issue is this for you?

**Comment by afilini** [comment]
In our specific case we can work around it because we are root in a container and we can just delete `/etc/electrs/config.toml` before doing everything else.

The problem is that we are trying to integrate this into https://github.com/RCasatta/electrsd/ which is basically a crate to spawn instances of electrs to run tests and whatnot. Since we can't control the environment in which that crate will run we would like to isolate it as much as possible. Basically, make sure that nothing else is loaded from the outside world other than the options we specify manually.

**Comment by Kixunil** [comment]
Interesting project! It seems that maybe for your use case the ability to avoid default files would be more useful than fixing this specific bug which will become irrelevant when `electrs` finally removes `cookie`? I believe I started writing that code some time ago so maybe I could dig up that branch and finish it.

**Comment by afilini** [comment]
Yes, an alternative solution could be be to have an option to ignore the default config files. That would solve our problem at least
